File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0
        end = len(numbers)-1

        while start < end:
            if target < numbers[start] + numbers[end]:
                end -= 1
            elif target > numbers[start] + numbers[end]:
                start += 1
            else:
                return [start+1, end+1]

        

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
#include<map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> m;
        for (int i = 0; i < size(nums); ++i){
            if (!m.contains(target-nums[i])) {
                m[nums[i]] = i;
            }
            else return {m[nums[i]], i};
        }
    }
};
